packages/trader/src/Modules/Trading/Containers/trade-params.jsx

```diff
@@ -11,28 +12,44 @@ import TakeProfit from 'Modules/Trading/Components/Form/TradeParams/Multiplier/t
 import Expiration from 'Modules/Trading/Components/Form/TradeParams/Multiplier/expiration.jsx';
 import { connect } from 'Stores/connect';
 
-const TradeParams = ({ form_components, is_minimized }) => {
+const TradeParams = ({ form_components, is_minimized, is_turbos }) => {
+    const [is_barriers_table_expanded, setIsBarriersTableExpanded] = React.useState(false);
+
     const isVisible = component_key => {
         return form_components.includes(component_key);
     };
+
+    const toggleBarriersTable = () => setIsBarriersTableExpanded(!is_barriers_table_expanded);
+
     return (
         <React.Fragment>
             {isVisible('duration') && <Duration key={'duration'} is_minimized={is_minimized} />}
-            {isVisible('barrier') && <Barrier key={'barrier'} is_minimized={is_minimized} />}
+            {isVisible('barrier') && (
+                <Barrier key={'barrier'} is_minimized={is_minimized} toggleBarriersTable={toggleBarriersTable} />
+            )}
             {isVisible('last_digit') && <LastDigit key={'last_digit'} is_minimized={is_minimized} />}
             {isVisible('amount') && <Amount key={'amount'} is_minimized={is_minimized} />}
             {isVisible('take_profit') && <TakeProfit key={'take_profit'} />}
             {isVisible('stop_loss') && <StopLoss key={'stop_loss'} />}
             {isVisible('cancellation') && <CancelDeal key={'cancellation'} />}
             {isVisible('expiration') && <Expiration key={'expiration'} />}
+            {isVisible('barrier') && is_turbos && is_barriers_table_expanded && (
+                <BarriersTable
+                    key={'barriers_table'}
+                    toggleBarriersTable={toggleBarriersTable}
+                    is_barriers_table_expanded={is_barriers_table_expanded}
+                />
+            )}
```

Comment:
```suggestion
            {isVisible('barrier') && <Barrier key={'barrier'} is_minimized={is_minimized} />}
            {isVisible('barriers_table') && <BarriersTable key={'barriers_table'} />
            {isVisible('last_digit') && <LastDigit key={'last_digit'} is_minimized={is_minimized} />}
            {isVisible('amount') && <Amount key={'amount'} is_minimized={is_minimized} />}
            {isVisible('take_profit') && <TakeProfit key={'take_profit'} />}
            {isVisible('stop_loss') && <StopLoss key={'stop_loss'} />}
            {isVisible('cancellation') && <CancelDeal key={'cancellation'} />}
            {isVisible('expiration') && <Expiration key={'expiration'} />}
```

@kate-deriv please remove local state from this file at line 16 and also toggleBarriersTable() at line 22, we need to add 'barriers_table' key to the config for Turbos in **packages/shared/src/utils/constants/contract.tsx** instead of 'barrier' in components array. Our new component `BarriersTable` should contain this local state for switching between expanded/collapsed (is_barriers_table_expanded) modes. For **collapsed mode**, you can add a <div like the one you are clicking now to expand the barriers table inside BarriersTable, please also add the current barrier value into this div and add Barrier title, and for expanded mode, you use the **expanded code** you currently have for BarriersTable. So everything will be incapsulated in BarriersTable🙂

Reply:
Done, thank you